pandas/io/formats/html.py

```diff
@@ -320,9 +320,15 @@ def _write_header(self, indent):
 
         self.write('</thead>', indent)
 
+    def _get_formatted_values(self):
```

Comment:
this is an api change yes?

Reply:
not at the moment, but probably needs to be if we want to have the option to truncate html output.

it is a change of behavior though. whether the current behavior is considered a bug or not.

Reply:
sorry, that's what I meant, this will *not* truncate by-default (but it IS truncating now)

Reply:
yes for to_html() only.

the behavior is unchanged for _repr_html_() (a.k.a. to_html(notebook=True) see #23820

Reply:
i was thinking that a change to the api should be part of this PR see https://github.com/pandas-dev/pandas/issues/17004#issuecomment-455821028.

but if we consider `display.max_colwidth` as a _display_ option and `to_html()` as _output_ then i think it is safe to consider #17004 as a bug and leave any api changes out of this PR.

this makes sense considering https://github.com/pandas-dev/pandas/pull/24840#issuecomment-455825159 regarding RC and api changes

Reply:
Should we wait for 0.24.1 on this?

Reply:
considering `display.max_rows` and `display.max_columns` are not used for `to_html(max_rows=None, max_cols=None)` then i see no reason why using the`display.max_colwidth` in `to_html()` as a default should not be labelled as a bug and rectified as soon as.

changing the api to _allow_ `display.max_colwidth` to be used in `to_html()` should be considered an enhancement and can wait for a later release.

Reply:
> this _does_ respect the max_colwith now, this is an API change here

@jreback  i'm confused! by _now_ do you mean in the master _now_ or with the changes _now_?

`display.max_rows` and `display.max_columns` are not used as default values in `to_html()`, so why should `display.max_colwidth` be used by default?

it _is_ a change of behaviour, granted. it is a bug fix. it is not a change to the api.

Reply:
yes the `test_ignore_display_max_colwidth` test will fail. the `test_display_max_colwidth` test is a regression test to check that the notebook display is unaffected by the change. this test passes with and without the code changes.

Reply:
While this is a bug fix, it'll change the behavior, in a way that I think is possibly to large for the RC.

I'd recommend waiting for 0.24.1.

Reply:
@TomAugspurger 

> I'd recommend waiting for 0.24.1

OK. so what's involved. just leave the PR open?

Reply:
@TomAugspurger if you want to include this in 0.24.1, is there anything else for me to do here? or would it be best to flag this as 0.25?

Reply:
Let's see what happens to https://github.com/pandas-dev/pandas/issues/24949.

I'm hopefully doing a 0.24.1 in the next day or two that's essentially just regression fixes. If we go with that, then this would be appropriate for 0.24.2 (sorry for bumping it repeatedly).

We don't have a 0.24.2.rst yet, but I'll make one if we plan to do a 0.24.1 release soon.

Reply:
if its not an api change, why wouldn't this go in 0.24.1 then?

Reply:
> 0.24.2 is the _same_ as 0.24.1 in that its meant for bug fixes. If its an api change then its 0.25, which is what I marked, can you move the whatsnew note.

@jreback: sorry that i'm creating unnecessary confusion. my understanding is that 0.24.1 is for regressions only. this PR is a bug fix and being concerned about potentially introducing a new regression, would prefer to see it go in 0.24.2.

The mention of 0.25 is because the fix could be seen as an unexpected change for a users perspective. This PR is not an api change.